internal/app/connectconformance/connectconformance.go

```diff
@@ -115,10 +115,13 @@ func Run(flags *Flags, logPrinter internal.Printer, errPrinter internal.Printer)
 	}
 
 	results, err := run(configCases, knownFailing, knownFlaky, runPatterns, skipPatterns, allSuites, logPrinter, errPrinter, flags)
-	if err != nil {
+	if results == nil {
 		return false, err
 	}
-	return results.report(logPrinter), nil
+	if err != nil {
+		errPrinter.Printf("%v", err)
+	}
+	return results.report(logPrinter) && err == nil, nil
```

Comment:
This is so that we can print the results, even after the client times out or crashes. We only return an error to the caller when there's an error _and_ no results. If we have both, report the error and the results. The bool value that is returned indicates the process should exit non-zero. So now we have to tell the caller to exit zero only when the results indicate no failures _and_ the error is nil.